### PR TITLE
Improve Code Organization

### PR DESCRIPTION
**Pull request by smizell**
This improves the organization of the code by breaking things into separate files. This also updates to the latest version of Uptown, which does not provide super methods because there was an issue with subclasses three levels deep.

### CODE REVIEW

File: lib/base.js [hunk 1]
```diff
@@ -113,719 +112,17 @@ var ElementRegistry = createClass({
...
+// any circuluar dependencies. BaseElement is also defined above in order to
+// satisfy any linting issues.
+BaseElement = require('./primitives/base-element')(registry);
+var NullElement = require('./primitives/null-element')(BaseElement);
+var StringElement = require('./primitives/string-element')(BaseElement);
+var NumberElement = require('./primitives/number-element')(BaseElement);
+var BooleanElement = require('./primitives/boolean-element')(BaseElement);
+var ArrayElement = require('./primitives/array-element')(BaseElement, registry);
+var MemberElement = require('./primitives/member-element')(BaseElement, registry);
+var ObjectElement = require('./primitives/object-element')(BaseElement, ArrayElement, MemberElement);
```
Comment (smizell) [inline lib/base.js:125]: These lines are mainly the biggest change. Dependencies injected when the library is first required.

**Comment by tu1ly** [comment]
@smizell looks ok (i haven't compared tests ;) )

**Comment by smizell** [comment]
@tu1ly Tests are just broken up into files. They use the main Minim API, so they should be OK IMO

**Comment by smizell** [comment]
@tu1ly Comments addressed

**Comment by tu1ly** [comment]
@smizell good to go

**Pull request merged**